src/performance.js

```diff
@@ -58,8 +66,48 @@ export class Performance {
 
     /** @const @private {!Array<TickEventDef>} */
     this.events_ = [];
+
+    /** @private {boolean} */
+    this.isViewerVisible_ = false;
+
+    this.win.addEventListener('unload', this.flush.bind(this));
```

Comment:
I don't think we will ever be getting `unload` events on mobile, but still should support. However, document's `visibilitychange` or viewer's `visibilitychange` should be good points for flush. See document-state/viewer APIs.